executor/load_data.go

```diff
@@ -99,8 +99,10 @@ func setNonRestrictiveFlags(stmtCtx *stmtctx.StatementContext) {
 	// TODO: DupKeyAsWarning represents too many "ignore error" paths, the
 	// meaning of this flag is not clear. I can only reuse it here.
 	stmtCtx.DupKeyAsWarning = true
-	stmtCtx.TruncateAsWarning = true
 	stmtCtx.BadNullAsWarning = true
+
+	oldFlags := stmtCtx.TypeConvContext.Flags()
+	stmtCtx.TypeConvContext = stmtCtx.TypeConvContext.WithFlags(oldFlags.WithTruncateAsWarning(true))
```

Comment:
Maybe we also need a new method `stmtCtx.UpdateTypeFlags`:

```
func (sc *StatementContext) UpdateTypeFlags(fn func(types.Flags) types.Flags) {
    flags := fn(stmtCtx.TypeConvContext.Flags())
    sc.TypeConvContext = sc.TypeConvContext.WithFlags(flags)
}

```

Reply:
IMO none of them is significantly better than the others 🤔 .

Following the thought of `UpdateTypeFlags`, we can have some options like:

```go
func (sc *StatementContext) UpdateTypeFlags(fns ...func(types.Flags) types.Flags) {
    var flags types.Flags
    for _, fn := range fns {
        flags := fn(stmtCtx.TypeConvContext.Flags())
    }
    sc.TypeConvContext = sc.TypeConvContext.WithFlags(flags)
}

// in types

func SetXXX(old types.Flags) types.Flags {
    return old.WithXXX(true)
}
func UnsetXXX(old types.Flags) types.Flags {
    return old.WithXXX(false)
}
```

But it seems to be more complex and cannot be inlined 🤦 .